src/js/components/modals/KillPodInstanceModal.js

```diff
@@ -0,0 +1,159 @@
+import {Confirm} from 'reactjs-components';
+import mixin from 'reactjs-mixin';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import MarathonStore from '../../stores/MarathonStore';
+import Pod from '../../structs/Pod';
+import StringUtil from '../../utils/StringUtil';
+
+const METHODS_TO_BIND = [
+  'handleButtonConfirm'
+];
+
+const ACTION_DISPLAY_NAMES = {
+  kill: 'Kill',
+  killAndScale: 'Kill and Scale'
+};
+
+class KillPodInstanceModal extends mixin(StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.store_listeners = [
+      {
+        name: 'marathon',
+        events: ['podInstanceKillSuccess', 'podInstanceKillError'],
+        suppressUpdate: true
+      }
+    ];
+
+    this.state = {
+      pendingRequest: false
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.open !== nextProps.open) {
+      this.setState({
+        errorMsg: null
+      });
+    }
+  }
+
+  shouldForceAction(message = this.state.errorMsg) {
+    return message && /force=true/.test(message);
+  }
+
+  handleButtonConfirm() {
+    let {pod, selectedItems} = this.props;
+    let killIDs = selectedItems.map(function (item) {
+      return item.id;
+    });
+
+    MarathonStore.killPodInstances(
+      pod,
+      killIDs,
+      this.shouldForceAction()
+    );
+    this.setState({pendingRequest: true});
+  }
+
+  onMarathonStorePodInstanceKillSuccess() {
+    this.setState({pendingRequest: false});
+    this.props.onClose();
+    this.props.onSuccess();
+  }
+
+  onMarathonStorePodInstanceKillError(errorMsg) {
+    this.setState({
+      pendingRequest: false,
+      errorMsg
+    });
+  }
+
+  getErrorMessage() {
+    let {errorMsg} = this.state;
+
+    if (!errorMsg) {
+      return null;
+    }
+
+    if (this.shouldForceAction(errorMsg)) {
+      return (
+        <h4 className="text-align-center text-danger flush-top">
+          App is currently locked by one or more deployments. Press the button
+          again to forcefully change and deploy the new configuration.
+        </h4>
+      );
+    }
+
+    return (
+      <p className="text-danger flush-top">{errorMsg}</p>
+    );
+  }
+
+  getModalContents() {
+    let selectedItemsLength = this.props.selectedItems.length;
+    let action = ACTION_DISPLAY_NAMES[this.props.action] || '';
```

Comment:
I thought of that but I intentionally wanted the UI to fail if the `action=` is wrong. Perhaps put some debug-time exception throwing instead of falling back to default.